### PR TITLE
wxHtmlEasyPrinting: allow to skip printing dialog

### PR DESCRIPTION
**Pull request by pavel-t**

### CODE REVIEW

**Comment by vadz** [comment]
Looks good to me, thanks!

I'll merge it if there are no comments from anybody else.

File: include/wx/html/htmprint.h [hunk 1]
```diff
@@ -296,6 +307,8 @@ class WXDLLIMPEXP_HTML wxHtmlEasyPrinting : public wxObject
     wxString m_Headers[2], m_Footers[2];
     wxWindow *m_ParentWindow;
 
+    DialogMode m_dialog;
```
Comment (PBfordev) [inline include/wx/html/htmprint.h]: Sorry for nitpicking, but wouldn't be better to name the variable `m_dialogMode` instead?

File: src/html/htmprint.cpp [hunk 2]
```diff
@@ -709,7 +711,10 @@ bool wxHtmlEasyPrinting::DoPrint(wxHtmlPrintout *printout)
     wxPrintDialogData printDialogData(*GetPrintData());
     wxPrinter printer(&printDialogData);
 
-    if (!printer.Print(m_ParentWindow, printout, true))
+    bool dialog = m_dialog != DIALOG_NEVER;
+    if (m_dialog == DIALOG_ONCE) m_dialog = DIALOG_NEVER;
```
Comment (PBfordev) [inline src/html/htmprint.cpp]: Some more nitpicking - the assignment statement should not be at the same line as the `if` expression.

Comment (vadz) [reply src/html/htmprint.cpp]: Yes, these are good points, thanks.

Comment (pavel-t) [reply src/html/htmprint.cpp]: Fixed.

**Comment by vadz** [comment]
Sorry, after looking at this again, I decided that we should rather call this "prompt mode", for consistency with `wxPrinter::Print()` parameter of the same name and because "dialog mode" is even more generic. So I've finally done a few changes to this PR before merging it, hopefully you don't mind, but please let me know if you do.

Thanks again!

**Pull request closed without merging**